database/cassandra/cassandra.go

```diff
@@ -296,16 +296,71 @@ func (c *Cassandra) ensureVersionTable() (err error) {
 		}
 	}()
 
-	err = c.session.Query(fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (version bigint, dirty boolean, PRIMARY KEY(version))", c.config.MigrationsTable)).Exec()
+	err = c.createVersionTable()
 	if err != nil {
 		return err
 	}
 	if _, _, err = c.Version(); err != nil {
+		var count int8
+		err = c.session.Query(`SELECT COUNT(*) FROM system_schema.columns WHERE keyspace_name = '` + c.config.KeyspaceName + `' AND table_name = '` + c.config.MigrationsTable + `' AND column_name = 'dummy'`).Scan(&count)
+		if err != nil {
+			return err
+		}
+
+		if count == 0 {
+			if err = c.upgradeVersionTable(); err != nil {
+				return err
+			}
+
+			if _, _, err = c.Version(); err != nil {
+				return err
+			}
+
+			return nil
+		}
 		return err
 	}
 	return nil
 }
 
+// upgradeVersionTable is responsible for upgrading legacy migrate users from the
+// older version table schema to the new one without disruption.
+// See: https://github.com/golang-migrate/migrate/issues/455
+func (c *Cassandra) upgradeVersionTable() (err error) {
+	version := -1
+	dirty := false
+	skip := false
+
+	query := `SELECT version, dirty FROM "` + c.config.MigrationsTable + `" LIMIT 1`
+	if err = c.session.Query(query).Scan(&version, &dirty); err != nil {
+		if err == gocql.ErrNotFound {
+			skip = true
```

Comment:
> failing to fetch the previous applied migration version

Given the previous implementation, the table may have been left in an empty state since the table is always emptied when `SetVersion` is called, but only filled back if a condition is met (and if it is not, no error is returned). This is why I don't error if it can't find previous version and instead just skip adding back the data

> Also, the previous version and status should be logged so subsequent failures can be manually recovered.

That's a valid point, I'll add some logging